feat(doc): add an example for a golang application deployment (#272)
CleverCloud/documentation: content/doc/applications/golang/_index.md

- `Module`: one or more packages you can `install`, defined in a `go.mod` file (`go.sum` for checksums)
- `Workspace`: one or more modules seamlessly combined, defined in a `go.work` file

> A module can be installed locally or from a remote repository if you pass its URL to the `install` command. A `Makefile` is sometimes used to define how a Go project is built, run and/or cleaned. The lightest form of a Go project is a `main.go` file to build. Some years ago, the `src/` folder was often used for source code, but using the `cmd/` folder instead is now a common practice.

Reviewer: ```suggestion
Install any module locally or from a remote repository by passing its URL to the `install` command. A `Makefile` is sometimes used to define how to build, run and/or clean a Go project. The lightest form of a Go project is a `main.go` file to build. The `src/` folder was often used for source code, but using the `cmd/` folder instead is now a common practice.
```
